modules/mobile_session.lua

```diff
@@ -1,430 +1,115 @@
-        end
+function mt.__index:StartRPC()
+  return self.mobile_session_impl:StartRPC()
+end
 
-        function d.CloseSession()
-          if self.heartbeatEnabled then
-            self:StopService(7)
-            self.test:FailTestCase("SDL didn't send anything for " .. self.heartbeatFromSDLTimer:interval()
-              .. " msecs. Closing session # " .. self.sessionId)
-          end
-        end
+function mt.__index:StopRPC()
+  self.mobile_session_impl:StopRPC()
+end
 
-        self.connection:OnInputData(function(_, msg)
-            if self.sessionId ~= msg.sessionId then return end
-            if self.heartbeatEnabled then
-                if msg.frameType == constants.FRAME_TYPE.CONTROL_FRAME and
-                   msg.frameInfo == constants.FRAME_INFO.HEARTBEAT_ACK and
-                   self.ignoreHeartBeatAck then
-                    return
-                end
-                self.heartbeatFromSDLTimer:reset()
-            end
-          end)
-        self.connection:OnMessageSent(function(sessionId)
-            if self.heartbeatEnabled and self.sessionId == sessionId then
-              self.heartbeatToSDLTimer:reset()
-            end
-          end)
-        qt.connect(self.heartbeatToSDLTimer, "timeout()", d, "SendHeartbeat()")
-        qt.connect(self.heartbeatFromSDLTimer, "timeout()", d, "CloseSession()")
-        self:StartHeartbeat()
-      end
+function mt.__index:Send(message)
+  -- Workaround
+    if message.serviceType == 7 then 
+      self.mobile_session_impl.rpc_services:CheckCorrelationID(message)
+    end
+  -- 
+  self.mobile_session_impl:Send(message)
+  return message
+end
 
-      local correlationId = self:SendRPC("RegisterAppInterface", self.regAppParams)
-      self:ExpectResponse(correlationId, { success = true })
-    end)
+function mt.__index:Start()
+  self.mobile_session_impl:Start()
 end
 
 function mt.__index:Stop()
-  self:StopService(7)
+  self.mobile_session_impl:Stop()
 end
 
+
 function module.MobileSession(test, connection, regAppParams)
   local res = { }
-  res.test = test
-  res.regAppParams = regAppParams
-  res.connection = connection
-  res.exp_list = test.expectations_list
-  res.messageId = 1
-  res.sessionId = 0
   res.correlationId = 1
-  res.version = config.defaultProtocolVersion or 2
-  res.hashCode = 0
-  res.heartbeatEnabled = true
-  res.sendHeartbeatToSDL = true
-  res.answerHeartbeatFromSDL = true
-  res.ignoreHeartBeatAck = false
   res.cor_id_func_map = { }
+  res.sessionId = 0
+
+
+  res.SessionId = {}
+  function res.SessionId.set(val) 
+    res.sessionId = val 
+  end
+  function res.SessionId.get() 
+    return res.sessionId
+  end
+
+
+  res.CorrelationId = {}
+  function res.CorrelationId.set(val) res.correlationId = val end
+  function res.CorrelationId.get() 
+    return  res.correlationId 
+  end
+
+
+  res.mobile_session_impl = mobileSessionImpl.MobileSessionImpl(
+  res.SessionId, res.CorrelationId, res.cor_id_func_map, test, connection, regAppParams)
```

Comment:
@\I cant see usage of cor_id_func_map in this class, please move it to MobileSessionImpl